os/drivers/video/video_null_driver.c

```diff
@@ -237,6 +262,87 @@ struct v4l2_mapping_s ctrl_mapping[] = {
 /****************************************************************************
  * Private Functions
  ****************************************************************************/
+
+/****************************************************************************
+ * Name: video_null_set_worker_delay
+ *
+ * Description:
+ *   This function calculates the delay requred for the worker thread
+ *   to invoke a callback based on the resolution chosen and the
+ *   video frames-per-second required.
+ *
+ * Input Paramaters:
+ *   priv - pointer to null_priv_t object holding the frame
+ *   resolution attributes.
+ *
+ * Returned Values:
+ *   OK on success, appropriate error code otherwise.
+ ****************************************************************************/
+static int video_null_set_worker_delay(FAR null_priv_t *priv)
+{
+	int i;
+	struct v4l2_format_s *format;
+	if (priv == NULL) {
+		videodbg("Unable to set worker delay duration!!!\n");
+	}
+
+	format = priv->cur_format;
```

Comment:
In case a format change change request is made while the device is already active (streaming), the device stream first has to be stopped before any format or frame parameters for the stream can be changed. So we need not worry about format changes after the lower driver has already started work.

Reply:
Video layer provides some apis to changes format, for example video_set_parm, it seems that can be called in the middle of streaming.

Reply:
Pending, re-checking.